src/sparsezoo/api/query_parser.py

```diff
@@ -93,12 +93,64 @@ def _parse_arguments(self) -> None:
 
     def _parse_fields(self) -> None:
```

Comment:
YMMV, but you might consider moving the `self.fields = ...` side effect of this function up to the caller or updating the method name to give a clue of the side effect. Maybe something like `_parse_and_set_fields` or something. I think I usually try to move the assignment up to the caller because I struggle to find a name for a method that does something and then makes an assignment. It can often be good to limit the number of functions that make assignments to make the lifecycle of the object easier to reason about.